packages/block-library/src/term-template/edit.js

```diff
@@ -0,0 +1,479 @@
+		</li>
+	);
+}
+
+/**
+ * Checks if a term is the currently active term.
+ *
+ * @param {number} termId               The term ID to check.
+ * @param {number} activeBlockContextId The currently active block context ID.
+ * @param {Array}  blockContexts        Array of block contexts.
+ * @return {boolean} True if the term is active, false otherwise.
+ */
+function isActiveTerm( termId, activeBlockContextId, blockContexts ) {
+	return termId === ( activeBlockContextId || blockContexts[ 0 ]?.termId );
+}
+
+export default function TermTemplateEdit( {
+	clientId,
+	attributes,
+	setAttributes,
+	context: {
+		termQuery: {
+			taxonomy,
+			order,
+			orderBy,
+			hideEmpty,
+			hierarchical,
+			parent,
+			perPage = 10,
+		} = {},
+	},
+	__unstableLayoutClassNames,
+} ) {
+	const [ activeBlockContextId, setActiveBlockContextId ] = useState();
+	const { replaceInnerBlocks } = useDispatch( blockEditorStore );
+	const { type: layoutType } = attributes?.layout || {};
+
+	// Switch to list if hierarchical is true and grid is selected.
+	useEffect( () => {
+		if ( hierarchical && layoutType === 'grid' ) {
+			setAttributes( {
+				layout: { type: 'default' },
+			} );
+		}
+	}, [ hierarchical, layoutType, setAttributes ] );
+
+	const queryArgs = {
+		order,
+		orderby: orderBy,
+		hide_empty: hideEmpty,
+		// To preview the data the closest to the frontend, we fetch the largest number of terms
+		// and limit them during rendering. This is because WP_Term_Query fetches data in hierarchical manner,
+		// while in editor we build the hierarchy manually. It also allows us to avoid re-fetching data when max terms changes.
+		per_page: 100,
+	};
+
+	const { records: terms, isResolving } = useEntityRecords(
+		'taxonomy',
+		taxonomy,
+		queryArgs
+	);
+
+	// Filter to show only top-level terms if "Show only top-level terms" is enabled.
+	const filteredTerms = useMemo( () => {
+		if ( ! terms || parent !== 0 ) {
+			return terms;
+		}
+		return terms.filter( ( term ) => ! term.parent );
+	}, [ terms, parent ] );
+
+	const { blocks, variations, defaultVariation } = useSelect(
+		( select ) => {
+			const { getBlocks } = select( blockEditorStore );
+			const { getBlockVariations, getDefaultBlockVariation } =
+				select( blocksStore );
+
+			return {
+				blocks: getBlocks( clientId ),
+				variations: getBlockVariations( 'core/term-template', 'block' ),
+				defaultVariation: getDefaultBlockVariation(
+					'core/term-template',
+					'block'
+				),
+			};
+		},
+		[ clientId ]
+	);
+
+	const blockProps = useBlockProps( {
+		className: __unstableLayoutClassNames,
+	} );
+
+	const blockContexts = useMemo(
+		() =>
+			filteredTerms?.map( ( term ) => ( {
+				taxonomy,
+				termId: term.id,
+				classList: `term-${ term.id }`,
+				termData: term,
+			} ) ),
+		[ filteredTerms, taxonomy ]
+	);
+
+	// Show variation picker if no blocks exist.
+	if ( ! blocks?.length ) {
+		return (
+			<div { ...blockProps }>
+				<BlockVariationPicker
+					icon="layout"
```

Comment:
```suggestion
					icon={ layout }
```
Need to pass an icon SVG and not a string, otherwise the dashicon will be displayed.